File: testdome/golang/numeric-input.go
package main
import (
	"fmt"
	"unicode"
)
type UserInput interface {
	Add(rune)
	GetValue() string
}

type NumericInput struct {
	input string
}

func (char *NumericInput) Add(input rune) {
	if unicode.IsDigit(input) {
		char.input += string(input)
	}
}

func (char *NumericInput) GetValue() string {
	return char.input
}

func main() {
  var input UserInput = &NumericInput{}
  input.Add('1')
  input.Add('a')
  input.Add('0')
  fmt.Println(input.GetValue())
}

File: testdome/python/numeric-input.py
#!/usr/bin/env python

class TextInput:
  def __init__(self):
    self.vals = ""
  def add(self,character):
    self.vals+=character
  def get_value(self):
    return self.vals

class NumericInput(TextInput):
  def __init__(self):
    super().__init__()
  def add(self,character):
    if character.isdigit(): self.vals+=character

if __name__ == '__main__':
  input = NumericInput()
  input.add("1")
  input.add("a")
  input.add("0")
  print(input.get_value())